Resource azurerm_cdn_endpoint_custom_domain gets created but not included in tfstate

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.0.3

### AzureRM Provider Version

3.10.0

### Affected Resource(s)/Data Source(s)

azurerm_cdn_endpoint_custom_domain

### Terraform Configuration Files

```hcl
resource "azurerm_cdn_endpoint_custom_domain" "cd" {
  for_each = toset(var.tenants)
  name            = "${var.aplicacao}-${var.ambiente}-cd-${each.value}"
  cdn_endpoint_id = azurerm_cdn_endpoint.endpoint.id
  host_name       = "${each.value}.conversaoform.${var.dominio}"
  cdn_managed_https {
    certificate_type = "Dedicated"
    protocol_type = "ServerNameIndication"
  }
}
```


### Debug Output/Panic Output

```shell
Terraform will perform the following actions:

  # azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"] will be created
  + resource "azurerm_cdn_endpoint_custom_domain" "cd" {
      + cdn_endpoint_id = "/subscriptions/<confidential>/resourceGroups/conversaoform-prod-rg/providers/Microsoft.Cdn/profiles/conversaoform-prod-cp/endpoints/conversaoform-prod-ce"
      + host_name       = "csvp-rj.conversaoform.mycompany.com"
      + id              = (known after apply)
      + name            = "my-cdn-customdomain-resname"

      + cdn_managed_https {
          + certificate_type = "Dedicated"
          + protocol_type    = "ServerNameIndication"
          + tls_version      = "TLS12"
        }
    }

.
.
.

azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Creating...
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [10s elapsed]
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [20s elapsed]
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [30s elapsed]
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [40s elapsed]
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [50s elapsed]
.
.
.
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [58m50s elapsed]
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [59m0s elapsed]
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [59m10s elapsed]
azurerm_cdn_endpoint_custom_domain.cd["csvp-rj"]: Still creating... [59m20s elapsed]
##[error]The operation was canceled.
Finishing: Aplicando plano de execução
```


### Expected Behaviour

Ideally, Azure would always create the resource (CDN endpoint custom domain) in a timely manner.
However, once in a while we get an occasion where it takes a lot of time. More than 1 hour, as the logs show.
When that happens, the pipeline job gets automatically canceled, stopping the Terraform process (that's an Azure DevOps Pipelines behaviour, as described [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?tabs=yaml&view=azure-devops#timeouts)).

When that happens, it would be awesome if either one of the following happened:

1. The resource creation on Azure got canceled as well and the tfstate never got that entry.
2. The resource creation on Azure went on (disconnected from the Terraform process) and the tfstate somehow indicated that.

Or perhaps, even better, would be if the resource declaration syntax had a parameter for "not waiting" the HTTPS configuration, which is clear for us that is the part Azure spends that time in.

### Actual Behaviour

The resource creation on Azure goes on (disconnected from the Terraform process) and the tfstate does not get any record of that resource. An inconsistent state: resource created on Azure but tfstate knows nothing about it.

As a consequence, the next time that pipeline runs, Terraform tries to create that resource again and, of course, we get a "resource with the ID ... already exists." error.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/cdn/cdn_endpoint_custom_domain_resource.go
+++ b/internal/services/cdn/cdn_endpoint_custom_domain_resource.go
@@ -168,6 +168,8 @@ func resourceArmCdnEndpointCustomDomainCreate(d *pluginsdk.ResourceData, meta in
 		return fmt.Errorf("waiting for creation of %q: %+v", id, err)
 	}
 
+	d.SetId(id.ID())
+
 	// Enable https if specified
 	var params cdn.BasicCustomDomainHTTPSParameters
 	if v, ok := d.GetOk("user_managed_https"); ok {
@@ -200,8 +202,6 @@ func resourceArmCdnEndpointCustomDomainCreate(d *pluginsdk.ResourceData, meta in
 			return err
 		}
 	}
-
-	d.SetId(id.ID())
 
 	return resourceArmCdnEndpointCustomDomainRead(d, meta)
 }

